### PR TITLE
Clawclaw

### PR DESCRIPTION
**Pull request by YHerm**

### CODE REVIEW

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java [hunk 1]
```diff
@@ -1,30 +1,23 @@
...
-    private NodeArm POST_CONE_DROP;
-    private NodeArm currentNode;
-    private int i = 1;
-    private final LinkedList<NodeArm> list = new LinkedList<>();
+    private RobotMap.TelescopicArm.PresetPositions currentNode;
+
+    protected final HashMap<RobotMap.TelescopicArm.PresetPositions, NodeArm> nodeMap = new HashMap<>();
     private static NodeBase instance;
-    private final double TOLERANCE_ANGLE = Units.degreesToRadians(3);
+    private  final double TOLERANCE_ANGLE = Units.degreesToRadians(3);
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java:18]: Ctrl + Alt + L this file

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java [hunk 2]
```diff
@@ -51,98 +38,69 @@ public NodeBase(){
...
+
+        nodeMap.get(CONE_HIGH).setClawPos(3);
+        nodeMap.get(CONE_MID).setClawPos(3);
+        nodeMap.get(CUBE_HIGH).setClawPos(3);
+        nodeMap.get(CUBE_MID).setClawPos(3);
+        nodeMap.get(INTAKE_GRAB_CONE_POSITION).setClawPos(1);
+        nodeMap.get(INTAKE_GRAB_CUBE_POSITION).setClawPos(2);
+        nodeMap.get(PRE_CONE_DROP).setClawPos(1);
+        nodeMap.get(POST_CONE_DROP).setClawPos(1);
+        nodeMap.get(LOW).setClawPos(3);
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java:76]: as well, change to Claw state

File: src/main/java/edu/greenblitz/tobyDetermined/commands/NodeToNeighbourSupplier.java [hunk 3]
```diff
@@ -11,25 +14,23 @@
...
-        SmartDashboard.putNumber("end node",end.getId());
-        LinkedList<NodeArm> path = AStar.getPath(start,end);
-        assert path != null;//can delete
-        if (path == null)//can delete
-            return null;//can delete
-        nodeCommands = new NodeToNeighbourCommand[path.size()-1];
-        for(int i = 0; i<path.size()-1; i++) {
-            nodeCommands[i] = new NodeToNeighbourCommand(path.get(i), path.get(i + 1));
+        SmartDashboard.putString("starting node", String.valueOf(start));
+        SmartDashboard.putString("end node", String.valueOf(end));
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/NodeToNeighbourSupplier.java:27]: move to arm debug dashboard

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/Dashboard.java [hunk 4]
```diff
@@ -155,7 +155,7 @@ public void armDashboard() {
 
 		armTab.addDouble("extender velocity",()-> Extender.getInstance().getVelocity());
 		armTab.addDouble("angular velocity", () -> Elbow.getInstance().getVelocity());
-		armTab.addDouble("Current Node ID", () -> NodeBase.getInstance().getCurrentNode().getId());
+		armTab.addString("Current Node ID", () -> String.valueOf(NodeBase.getInstance().getCurrentNode()));
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/Dashboard.java:158]: change dashboard display name (from current node id)

Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/Dashboard.java:158]: change from String.valueOf() to .toString()

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/tobyDetermined/commands/NodeToNeighbourSupplier.java (hunk 3 above)
Comment (YHerm) [reply src/main/java/edu/greenblitz/tobyDetermined/commands/NodeToNeighbourSupplier.java:27]: what where when how

**Comment by NoamRosenberg08** [comment]
no need anymore

**Pull request closed without merging**